### PR TITLE
Study View updates

### PR DESCRIPTION
**Pull request by kalletlak**
# What? Why?
Fix
1. https://github.com/cBioPortal/cbioportal/issues/4900
2. remove recurrent api call when there are no filter applied
3. https://github.com/cBioPortal/cbioportal/issues/4944
4. https://github.com/cBioPortal/cbioportal/issues/4928

### CODE REVIEW

File: src/pages/studyView/StudyViewPageStore.ts [hunk 1]
```diff
@@ -1107,15 +1107,23 @@ export class StudyViewPageStore {
                 } : DEFAULT_LAYOUT_PROPS.dimensions[chartType]);
             });
 
+            const cancerTypeIds = _.uniq(this.queriedPhysicalStudies.result.map(study=>study.cancerTypeId));
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: @jjgao we use the `mixed` in the cancer type id field to determine whether it is a mixed study. This should cover most of the use cases. But when user select one cancer type from a mixed study and saved as VS. Next time, when they open the VS, the survival plot should be shown by default?

Comment (jjgao) [reply src/pages/studyView/StudyViewPageStore.ts]: That's a good point. When opening a VS, is there a way to look at the main cancer type, if there is only one, show survival?

Alternatively, we don't have to worry about it. We will move the survival plots to comparisons tab later anyway.

Comment (zhx828) [reply src/pages/studyView/StudyViewPageStore.ts]: Let's postpone it.

File: src/pages/studyView/StudyViewPageStore.ts [hunk 2]
```diff
@@ -1930,7 +1948,15 @@ export class StudyViewPageStore {
                 }
 
                 if (UniqueKey.MUTATION_COUNT_CNA_FRACTION in this.chartMetaSet) {
-                    ret[UniqueKey.MUTATION_COUNT_CNA_FRACTION] = (this.mutationCountVsFractionGenomeAlteredData.result || []).length;
+                    const mutationCountVsFractionGenomeAlteredDataSet = this.mutationCountVsFractionGenomeAlteredDataSet.result;
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts]: would be nice to have a comment here explaining this

**Pull request merged**